src/Microsoft.ML.Core/Data/ColumnType.cs

```diff
@@ -478,42 +478,39 @@ public override bool Equals(ColumnType other)
     /// class numbers, etc. For example, in multi-class classification, the label is typically
     /// a class number which is naturally a KeyType.
     ///
-    /// KeyTypes can be contiguous (the class number example), in which case they can have
-    /// a cardinality/Count. For non-contiguous KeyTypes the Count property returns zero.
-    /// Any KeyType (contiguous or not) can have a Min value. The Min value is always >= 0.
+    /// KeyTypes have a cardinality/Count that is strictly positive.
     ///
     /// Note that the representation value does not necessarily match the logical value.
-    /// For example, if a KeyType has range 1000-5000, then it has a Min of 1000, Count
-    /// of 4001, but the representational values are 1-4001. The representation value zero
-    /// is reserved to mean none/invalid.
+    /// For example, if a KeyType has range 0-5000, then it has a Count of 5001, but
+    /// the representational values are 1-5001. The representation value zero is reserved
+    /// to mean none/invalid.
     /// </summary>
     public sealed class KeyType : PrimitiveType
     {
-        private KeyType(Type type, DataKind kind, ulong min, int count, bool contiguous)
+        private KeyType(Type type, DataKind kind, ulong count)
             : base(type, kind)
         {
             Contracts.AssertValue(type);
             Contracts.Assert(kind.ToType() == type);
-
-            Contracts.CheckParam(min >= 0, nameof(min));
-            Contracts.CheckParam(count >= 0, nameof(count), "Must be non-negative.");
-            Contracts.CheckParam((ulong)count <= ulong.MaxValue - min, nameof(count));
-            Contracts.CheckParam((ulong)count <= kind.ToMaxInt(), nameof(count));
-            Contracts.CheckParam(contiguous || count == 0, nameof(count), "Must be 0 for non-contiguous");
-
-            Contiguous = contiguous;
-            Min = min;
+            Contracts.CheckParam(0 < count && count <= kind.ToMaxInt(), nameof(count),
+                "The cardinality of a " + nameof(KeyType) + " must not exceed " + nameof(type) + ".MaxValue and must be strictly positive.");
```

Comment:
>nameof(type) [](start = 82, length = 12)

This will be `type.MaxValue`, but I think you really meant `type.Name`. Note that `nameof(type)` is literally going to just be the string `"type"`.